lib/src/view/user/user_screen.dart

```diff
@@ -13,9 +13,23 @@ import 'user_activity.dart';
 import 'user_profile.dart';
 
 class UserScreen extends ConsumerWidget {
-  const UserScreen({required this.user, super.key});
+  const UserScreen._({this.lightUser, this.user, super.key})
+      : assert(
+          (lightUser != null || user != null) &&
+              !(lightUser != null && user != null),
+          'Either lightUser or User should be provided but not both',
+        );
 
-  final LightUser user;
+  const factory UserScreen.fromUser({required User user, Key? key}) =
```

Comment:
I don't understand this change actually. 

Why adding more complexity? We should always load the user from ID. 

It looks like this change was made to handle the case where the user is not found in a profile link. But it is better to handle this case in the async error handler and show a "User not found" here instead.